internal/runtime/client/client.go

```diff
@@ -339,18 +352,25 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, fo
 		ignore := *registration.FailurePolicy == runtimev1.FailurePolicyIgnore
 		if _, ok := err.(errCallingExtensionHandler); ok && ignore {
 			// Update the response to a default success response and return.
+			log.V(5).Info(fmt.Sprintf("ignoring error calling extension handler because of FailurePolicy %q", *registration.FailurePolicy))
 			response.SetStatus(runtimehooksv1.ResponseStatusSuccess)
 			response.SetMessage("")
 			return nil
 		}
+		log.Error(err, "failed to call extension handler")
 		return errors.Wrapf(err, "failed to call extension handler %q", name)
 	}
 
 	// If the received response is a failure then return an error.
 	if response.GetStatus() == runtimehooksv1.ResponseStatusFailure {
+		log.Error(err, "extension handler returned a failure response")
 		return errors.Errorf("failed to call extension handler %q: got failure response with message %q", name, response.GetMessage())
 	}
 
+	if retryResponse, ok := response.(runtimehooksv1.RetryResponseObject); ok && retryResponse.GetRetryAfterSeconds() != 0 {
+		log.Info(fmt.Sprintf("extension handler returned blocking response with retryAfterSeconds of %q", retryResponse.GetRetryAfterSeconds()))
+	}
```

Comment:
Let's also log the success-case. Is it just an else here?

(same in CallAllExtensions)

Reply:
Yes we should be okay with that in an else case. That would capture both a `RetryResponseObject` being non-blocking and getting a success response for a simple `ResponseObject`. Logging a success response would be fine in both the cases.